tests/remotes/gdrive.py

```diff
@@ -1,41 +1,85 @@
 # pylint:disable=abstract-method
+import locale
 import os
+import pathlib
+import tempfile
 import uuid
 
 import pytest
 from funcy import cached_property
 
-from dvc.path_info import CloudURLInfo
-from dvc.tree.gdrive import GDriveTree
+from dvc.path_info import CloudURLInfo, PathInfo
+from dvc.tree.gdrive import GDriveTree, GDriveURLInfo
 
 from .base import Base
 
 TEST_GDRIVE_REPO_BUCKET = "root"
 
 
 class GDrive(Base, CloudURLInfo):
+    def __init__(self, url, dvc):
+        CloudURLInfo.__init__(self, url)
+        Base.__init__(self, url)
+        self.tree = GDriveTree(dvc, self.config)
+
     @staticmethod
     def should_test():
         return bool(os.getenv(GDriveTree.GDRIVE_CREDENTIALS_DATA))
 
     @cached_property
     def config(self):
-        return {
-            "url": self.url,
-            "gdrive_service_account_email": "test",
-            "gdrive_service_account_p12_file_path": "test.p12",
-            "gdrive_use_service_account": True,
-        }
+        if GDrive.should_test():
+            return {
+                "url": self.url,
+            }
+        else:
+            return {
+                "url": self.url,
+                "gdrive_service_account_email": "test",
+                "gdrive_service_account_p12_file_path": "test.p12",
+                "gdrive_use_service_account": True,
+            }
```

Comment:
This looks confusing, could you clarify what is going on?

Also no need for `else` if you return before it.

Reply:
My fault. Done!